misk-aws/src/main/kotlin/misk/jobqueue/sqs/AwsSqsJobHandlerModule.kt

```diff
@@ -11,38 +11,47 @@ import kotlin.reflect.KClass
 
 /**
  * Install this module to register a handler for an SQS queue,
- * along with its corresponding retry queue
+ * and if specified, registers its corresponding retry queue.
  */
 class AwsSqsJobHandlerModule<T : JobHandler> private constructor(
   private val queueName: QueueName,
-  private val handler: KClass<T>
+  private val handler: KClass<T>,
+  private val installRetryQueue: Boolean
 ) : KAbstractModule() {
   override fun configure() {
     newMapBinder<QueueName, JobHandler>().addBinding(queueName).to(handler.java)
-    newMapBinder<QueueName, JobHandler>().addBinding(queueName.retryQueue).to(handler.java)
+
+    if (installRetryQueue) {
+      newMapBinder<QueueName, JobHandler>().addBinding(queueName.retryQueue).to(handler.java)
+    }
+
     install(ServiceModule<AwsSqsJobHandlerSubscriptionService>())
   }
 
   companion object {
-    inline fun <reified T : JobHandler> create(queueName: QueueName):
-        AwsSqsJobHandlerModule<T> = create(queueName, T::class)
+    inline fun <reified T : JobHandler> create(
```

Comment:
this is not going to be backwards binary compatible with existing callers. just add an overloaded method, or tag with `@JvmOverloads`

Reply:
Got it. Done

Reply:
same for the other methods too

Reply:
Oops sorry. Added @JvmOverloads